### ISSUE TITLE
Neptune 3, and Neptune 2S

### ISSUE THREAD

**Issue by RickzFPV** [mlee12382/Neptune_3#3]
@mlee12382 Michael I am curious how I could obtain pinout diagrams for both the Neptune 2S and the Neptune 3 I am in the process of testing Klipper and Can't seem to find a decent pinout diagram for either of them.  they show what plug is which but not the pin designations.  Any help will be much appreciated Thank you.

**Comment by mlee12382** [comment]
All of the schematics and pinouts for the 2/2S/2D are in the source materials zip on elegoo's support page. https://www.elegoo.com/pages/3d-printing-user-support
Here's the N3 pinout, the pro uses a different board though and I don't have anything for it yet.
![Neptune3 Pinout.png](https://user-images.githubusercontent.com/89716126/208951705-421a8f0a-1bc9-476f-af78-1c3da89fc29f.png)

**Comment by RickzFPV** [comment]
Hook thank you @mlee12382 i didn’t know they were there their site is kind of hard to get around on at times thank you again

**Comment by mlee12382** [comment]
> Hook thank you @mlee12382 i didn’t know they were there their site is kind of hard to get around on at times thank you again

I wish they had as much documentation for the other printers as they do for the N2

**Comment by RickzFPV** [comment]
@mlee12382 do you think you will be upgrading your elegoo config stuff for Marlin 2.1.2 I am very interested in seeing how input shaper runs with Marlin

**Comment by mlee12382** [comment]
Yeah, I will get around to it at some point, been doing other things recently. I might take a look at it sometime in the next couple weekends if I have time.

**Comment by RickzFPV** [comment]
@mlee12382 Michael just wanted to let you know that I tried to get your 2.1.2 firmware going last night, and printer was acting kind of erratic getting it set up just printing a calibration cube it would start get through about 2 layers then nozzle would just leave the print go to the side of the bed and then come back and start printing again.  Tried to diagnose it but could not for the life of me get it figured out.  I am back on 2.1.1 and it is printing perfect, so I have to suspect the new firmware.  not sure if anyone else is having issues or erratic behavior with their printers, but just wanted to let you know.  I will just stick with the 2.1.1 for a bit until some things get worked out as I know it is still so new.

**Comment by mlee12382** [comment]
Interesting, I will check through the code after work and see if maybe I missed something. That's on your 2S?

**Comment by RickzFPV** [comment]
Yeah it is on my 2S

**Comment by RickzFPV** [comment]
@mlee12382 just curious if you know if any progress is being made with the Neptune 3 screen firmware and being able to run advanced settings menus yet.  been thinking about trying the new Marlin 2.1.2 on it, but the only way to do some of the advanced stuff on it is through octoprint or Pronterface and would like to do it from the screen.  been struggling a little bit with one of my 2S's bought it on a as is deal from a liquidation place and I think the right hand side X gantry plate is bent and Can't find any parts for these anywhere.  would really like to try linear rails on it but because of the 2040 extrusions elegoo uses they can't be used as far as I can tell.

**Comment by mlee12382** [comment]
Check out TheFeralEngineer on YouTube he has linear rail mods for the 3 that should also work with the 2. He also has 2.1.2 kind of working on the N3 but he had to use the MKS H43 screen instead of the stock screen.

**Comment by RickzFPV** [comment]
@mlee12382 Michal I am curious about something.  Been running your firmware on all my elegoo printers for a while and everything has been awesome.  I have noticed recently that I have been having to level the beds more on my Neptune 2s's, not sure if beds are warping because of use, or if it is something else.  My question is can I add manual bed levelling to your firmware with a BL Touch the reason why I want to do this is because I would like to level the bed without having to disable the steppers to see if it becomes more accurate.  I have been running my neptune 3 a ton lately and with your previous help it has been running flawless.

Rick

**Comment by mlee12382** [comment]
> @mlee12382 Michal I am curious about something.  Been running your firmware on all my elegoo printers for a while and everything has been awesome.  I have noticed recently that I have been having to level the beds more on my Neptune 2s's, not sure if beds are warping because of use, or if it is something else.  My question is can I add manual bed levelling to your firmware with a BL Touch the reason why I want to do this is because I would like to level the bed without having to disable the steppers to see if it becomes more accurate.  I have been running my neptune 3 a ton lately and with your previous help it has been running flawless.
> 
> Rick

You should be able to use the tramming wizard for manual leveling it uses the probe to help you adjust your leveling knobs. And if you're also running octoprint the plugin you want is AutoBim.

**Comment by RickzFPV** [comment]
Ok Perfect Thank You

**Comment by RickzFPV** [comment]
@mlee12382 I need your help figuring something out lol.  I replaced the hotend on 1 of my Neptune 2s's to a creality direct drive, the bl touch is now on the opposite side of the hotend.  I know my probe offsets are now X -44, and Y -9 but when I build the firmware it is erroring because it can't reach points.  I am not sure if I need to adjust the bed margins or not.  I have firmware ready except for this bump.

**Comment by mlee12382** [comment]
@RickzFPV it's probably the tramming points, there's lcd leveling and assisted tramming that you need to adjust the values for to match the probe offset in the firmware.

**Comment by mlee12382** [comment]
@RickzFPV   #define BED_TRAMMING_INSET_LFRB { 35, 35, 35, 35 } // (mm) Left, Front, Right, Back insets in Configuration.h and assisted tramming in configuration_adv.h

**Comment by RickzFPV** [comment]
@mlee12382 I have been trying since last night to build this firmware and I just can't grasp this Margins thing.  I have read everything I could understand on it and it just doesn't make sense.  Let me explain where I am now, I am just trying to modify your 2.1.2 for a 2s for this new hotend.  I had to change Y bed size so it doesn't bump the tensioner.  so bed size is 235x230, Probe offsets are -44, and -9, keeps giving me an error that it can't reach right side.  I even tried the same Margins as my ender 3 Pro and it gave same error, yet my ender 3 Pro works perfectly. I just don't understand what is wrong.

Rick

**Comment by mlee12382** [comment]
So I think in Configuration.h  bed_tramming_inset_lfrb  should be 35, 35, 45, 35
And in configuration_adv.h tramming_point_xy should be 35, 35 190, 35 190, 200 35, 200 and then I  think it should be able to compile.

**Comment by RickzFPV** [comment]
@mlee12382 Thank you so much will have this printer printing in about an hour now.  You are an absolute Genius when it comes to these elegoo printers.  Any word yet on a screen update for the neptune 3 I would really like to have advanced features on the touch screen.  

Rick

**Comment by mlee12382** [comment]
Tbh I've kinda abandoned that project, there's one or 2 people that have been playing around with the screens some but I'm not sure how far they've gotten or if they're even still working on it. 
I think at this point a lot of people that are trying to do the more advanced stuff are just switching to klipper and replacing the screen with something capable of running KlipperScreen. The N3 seems to be somewhat crippled by marlin and the wonky implementations from elegoo.

**Comment by RickzFPV** [comment]
Yeah I totally understand that.  I am still messing with a couple of these 2s’s in the stock configuration but I have 1 running on a newer Mks board with a better screen and 1 running on a BTT board with their screen and they run awesome.  I think elegoo has shifted all their focus on the pro series of their printers.  My neptune 3 with your firmware has turned into a work horse for me it just prints anytime I ask it too.
